public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+include ../_util-fns
+
+:marked
+  _Reactive forms_ is an Angular technology for creating data entry forms in a _reactive_ style.
+  This guide explains reactive forms as you follow the steps to build a "Hero Detail Editor" form.
+
+a#toc
+:marked
+  ## Contents
+
+  - [Introduction to reactive forms](#intro)
+  - [Setup](#setup)
+  - [Create a data model](#data-model)
+  - [Create a _reactive forms_ component](#create-component)
+  - [Create its template file](#create-template)
+  - [Import the _ReactiveFormsModule_](#import)
+  - [Update the app to use the form component](#update)
+  - [Introduction to _FormBuilder_](#formbuilder)
+  - [Group _FormControls_](#grouping)
+  - [Inspect _FormControl_ properties](#properties)
+  - [Set form model data using _setValue_ and _patchValue_](#set-data)
+  - [Use _FormArray_ to present an array of _FormGroups_](#form-array)
+  - [Watch control changes](#watch-control)
+  - [Save form data](#save)
+
+  Try the <live-example plnkr="final" title="Reactive Forms (final) in Plunker">Reactive Forms live-example</live-example>.
+
+  You can also run the <live-example title="Reactive Forms Demo in Plunker">Reactive Forms Demo</live-example> version
+  and choose one of the intermediate steps from the "demo picker" at the top.
+
+a#intro
+:marked
+  ## Introduction to Reactive Forms
+
+  Angular offers two form-building technologies: _reactive_ forms and _template-driven_ forms.
+  The two technologies belong to the `@angular/forms` library
+  and share a common set of form control classes.
+
+  But they diverge markedly in philosophy, programming style, and technique.
+  They even have their own modules: the `ReactiveFormsModule` and the `FormsModule`.
+
+  ### _Reactive_ forms
+  Angular _reactive_ forms facilitate a _reactive style_ of programming 
+  that favors explicit management of the data flowing between
+  a non-UI _data model_ (typically retrieved from a server) and a 
+  UI-oriented _form model_ that retains the states 
+  and values of the HTML controls on screen.
+
+  With _reactive_ forms, you create a tree of Angular form controls in code, and 
+  bind them to HTML widgets in the component template, using techniques described in this guide.
+
+  The component class is responsible for all data flowing between the component and the screen.
+  It has immediate access to both the data model and the form control structure. 
+  It pushes data model values into the form controls and 
+  pulls user-changed values back out of the form controls.
+  The component can _observe_ changes in form control state and _react_ to those changes.
+
+  In keeping with the reactive paradigm, the component should preserve the immutability of the _data model_,
+  treating it as a pure source of original values.
+  Rather than update the data model directly, 
+  the component extracts user changes and forwards them to an external component or service,
```

Comment:
Does this work? 

>   In keeping with the reactive paradigm, the component  
  preserves the immutability of the _data model_,
  treating it as a pure source of original values.
  Rather than update the data model directly, 
  the component extracts user changes and forwards them to an external component or service,
  which does something with them (such as saving them) 
  and returns a new _data model_ to the component that reflects the updated model state.

  Using reactive form directives does not require you to follow all reactive priniciples,
  but it does facilitate the reactive programming approach should you choose to use it.